src/kirby/components/icon/icon.component.ts

```diff
@@ -10,55 +13,67 @@ import { Component, Input, HostBinding } from '@angular/core';
     class: 'kirby-icon',
   },
 })
-export class IconComponent {
-  static DEFAULT_ICON_CODE = 0xf2cf;
-  static DEFAULT_ICON_NAME: 'cog' = 'cog';
+export class IconComponent implements OnChanges {
+  defaultIcon: CustomIcon = this.getCustomIcon(kirbyCustomIconSettings.icons, 'cog');
+  fontFamily: string = kirbyCustomIconSettings.fontfamily;
+  private _icon = (this.icon = this.defaultIcon);
+
+  @Input() name: string;
+  @Input() customName: string;
+
+  get icon(): CustomIcon {
+    return this._icon;
+  }
+
+  set icon(icon: CustomIcon) {
+    if (!this.customIconSettings && this.customName) {
+      console.warn(
+        'CUSTOM_FONT_SETTINGS provider in your module.ts is not set. Read documentation on how to set it up.'
+      );
+    }
+
+    if (!icon && (this.name || this.customName)) {
+      console.warn(`Icon with name "${this.name || this.customName}" was not found.`);
+      icon = this.defaultIcon;
 
-  @Input() name:
-    | 'add'
-    | 'close'
-    | 'cog'
-    | 'swap'
-    | 'move'
-    | 'log-out'
-    | 'more'
-    | 'arrow-back'
-    | 'help'
-    | 'attach'
-    | 'search'
-    | 'checkbox-outline'
-    | 'checkbox'
-    | 'menu'
-    | 'person' = IconComponent.DEFAULT_ICON_NAME;
+      if (!icon) {
+        console.warn('Default icon was not found.');
+        return;
+      }
+    }
 
-  getIonicIconName(name: string): string {
-    const icon = iconsCharCodeMap[name];
-    return icon !== undefined ? name : IconComponent.DEFAULT_ICON_NAME;
+    if (icon) {
+      this._icon = {
+        ...icon,
+        fromCharCode: this.fromCharCode(icon.unicode),
```

Comment:
Consider inlining (1 line method wrapper):
```suggestion
        fromCharCode: String.fromCharCode(icon.unicode),
```